test/test_framework.py

```diff
@@ -573,19 +573,66 @@ class CustomObject:
         # Assignment and keywords like del are not supported in lambdas so functions are used instead.
         test_funcs = [(
             lambda: setattr(obj.state, 'dict', {}),
-            lambda: self.assertEqual(dict(obj.state.dict), {})
+            lambda: self.assertEqual(obj.state.dict, {})
         ), (
             lambda: obj.state.dict.update({"a": {}}),
-            lambda: self.assertEqual(dict(obj.state.dict), {"a": {}})
+            lambda: self.assertEqual(obj.state.dict, {"a": {}})
         ), (
             lambda: obj.state.dict["a"].update({"b": "c"}),
-            lambda: self.assertEqual(dict(obj.state.dict), {"a": {"b": "c"}})
+            lambda: self.assertEqual(obj.state.dict, {"a": {"b": "c"}})
         ), (
             lambda: obj.state.dict["a"].update({"d": "e"}),
-            lambda: self.assertEqual(dict(obj.state.dict), {"a": {"b": "c", "d": "e"}})
+            lambda: self.assertEqual(obj.state.dict, {"a": {"b": "c", "d": "e"}})
         ), (
             lambda: obj.state.dict["a"].pop('d'),
-            lambda: self.assertEqual(dict(obj.state.dict), {"a": {"b": "c"}})
+            lambda: self.assertEqual(obj.state.dict, {"a": {"b": "c"}})
+        ), (
+            lambda: setattr(obj.state, 'list', []),
+            lambda: self.assertEqual(obj.state.list, [])
+        ), (
+            lambda: (
+                obj.state.list.append("a"),
```

Comment:
I didn't realize that each test case in the table was using the state left behind by the prior test. Is that indeed the case, and if so can we please take this chance to fix it so it doesn't do that?  It's a nightmare to maintain and update a test like that as it's just a long chain of changes that depend on each other in non-obvious ways. The equality test on the prior PR that went in was nicer in that regard, after Cory's suggestion. Maybe we can do something similar here?

Reply:
I agree, I suppose we could.

I made the change and pushed it.